spec/datadog/tracing/tracer_spec.rb

```diff
@@ -535,6 +535,26 @@
           end
         end
       end
+
+      context 'for span sampling' do
+        let(:tracer_options) { super().merge(span_sampler: span_sampler) }
+        let(:span_sampler) { instance_double(Datadog::Tracing::Sampling::Span::Sampler) }
+        let(:block) do
+          proc do |span_op, trace_op|
+            @span_op = span_op
+            @trace_op = trace_op
+          end
+        end
+
+        before do
```

Comment:
I hesitate to use `before` here because there is only one example.

The mock verification could simply be written as 
```
expect(span_sampler).to receive(:sample!).with(@trace_op, @span_op.send(:build_span))
```
instead of setting up spy and verifying it with `have_received`

Reply:
I don't think I can do that because @trace_op only exists during the execution of `sample!`. If I don't capture it right then, it's gone.

Reply:
As an additional remark: I moved it outside of the `it` this part reads as test setup, and IMO pollutes the test assertion.